pandas/core/ops/__init__.py

```diff
@@ -1716,10 +1717,32 @@ def wrapper(left, right):
             # Note: we cannot use dispatch_to_index_op because
             #  that may incorrectly raise TypeError when we
             #  should get NullFrequencyError
-            result = op(pd.Index(left), right)
-            return construct_result(
-                left, result, index=left.index, name=res_name, dtype=result.dtype
-            )
+            orig_right = right
+            if is_scalar(right):
+                # broadcast and wrap in a TimedeltaIndex
+                # TODO: possible perf improvement by using NaTD scalar
```

Comment:
I don't think this this is a worthwhile comment here as NaTD is not likely to happen

Reply:
I think there's a decent chance we'll implement it for internal use to de-duplicate logic like this.

Reply:
Removed comment